sql/field.cc

```diff
@@ -75,6 +75,14 @@ bool Field::marked_for_read() const
 
 bool Field::marked_for_write_or_computed() const
 {
+  /*
+    Field_new_decimal::store_value() will be called if the field is null when
+    binlog applier is unpacking a row image. In that case, write_set should not
+    be checked.
+  */
+  if (get_thd() && get_thd()->lex->is_stmt_row_injection())
+    return true;
```

Comment:
I think I got the idea of this hunk, let me share with what I understood.
At this point the rbr applier  executes a stack of 
```
Field::marked_for_write_or_computed
Field_new_decimal::store_value
Field_new_decimal::store_value
Field_new_decimal::reset
unpack_field
unpack_row
Rows_log_event::unpack_current_row
Rows_log_event::write_row
Write_rows_log_event::do_exec_row
Rows_log_event::do_apply_event
``` 
`Field_new_decimal::reset` does not need any bitmap as it just wipes of any old content. And it looks to be me that the above hunk assumes `Field::marked_for_write_or_computed` is indeed invoked necessarily from the `unpack_row` stack.
But can `Field_new_decimal::store_value -> Field::marked_for_write_or_computed`  be also reached later, that is after the resetting, e.g [here](https://github.com/MariaDB/server/blob/11.4/sql/rpl_record.cc#L232)?

Reply:
No, for applier row event Field::unpack() is called to store the value if it is not null. Field::store_value() is called only for reset() and it happens only for Field_decimal and Field_new_decimal, other type of fields don't call ::store_value() in reset().

Reply:
Thanks for correcting me, LiBing!
After thinking more and brief discussion with Bar I think it's safer to tackle assert in the replication code,
which I propose to in [a review commit](https://github.com/MariaDB/server/blob/review-11.4-MDEV-33049/sql/rpl_record.cc#L205) on top of yours.
Please check that out.

Reply:
@FooBarrior, let me also invite you as this bug is certainly about ` 3a42f2869e80` changes. 
Maybe you'll have the best of solutions :-)! 
Thank you at once!

Reply:
Hi Andrei, fixed as your suggestion.